packages/trader/src/App/Components/Elements/VideoPlayer/video-player.tsx

```diff
@@ -0,0 +1,270 @@
+import React from 'react';
+import { Stream, StreamPlayerApi } from '@cloudflare/stream-react';
+import { user_browser, mobileOSDetect } from '@deriv/shared';
+import VideoOverlay from './video-overlay';
+import VideoControls from './video-controls';
+
+type TVideoPlayerProps = {
+    src: string;
+    is_mobile?: boolean;
+    data_testid?: string;
+};
+
+const VideoPlayer = ({ src, is_mobile, data_testid }: TVideoPlayerProps) => {
+    const should_autoplay =
+        (!user_browser.isSafari() || (is_mobile && mobileOSDetect() !== 'iOS' && mobileOSDetect() !== 'unknown')) ??
+        true;
+
+    const [current_time, setCurrentTime] = React.useState<number>();
+    const [has_enlarged_dot, setHasEnlargedDot] = React.useState(false);
+    const [is_animated, setIsAnimated] = React.useState(true);
+    const [is_playing, setIsPlaying] = React.useState(false);
+    const [is_ended, setIsEnded] = React.useState(false);
+    const [is_muted, setIsMuted] = React.useState(false);
+    const [playback_rate, setPlaybackRate] = React.useState(1);
+    const [show_controls, setShowControls] = React.useState(!should_autoplay);
+    const [shift_X, setShiftX] = React.useState(0);
+    const [video_duration, setVideoDuration] = React.useState<number>();
+    const [volume, setVolume] = React.useState(0.5);
+
+    const video_ref = React.useRef<StreamPlayerApi>();
+    const progress_bar_filled_ref = React.useRef<HTMLDivElement>(null);
+    const progress_bar_ref = React.useRef<HTMLDivElement>(null);
+    const progress_dot_ref = React.useRef<HTMLSpanElement>(null);
+    const animation_ref = React.useRef(0);
+    const timer_ref = React.useRef<ReturnType<typeof setTimeout>>();
+    const is_dragging = React.useRef(false);
```

Comment:
Same here, can we have is_dragging as a state?

Reply:
Same here